### PR TITLE
[6.3]register twice with uppercase chars in hostname (BZ1361309)

### PR DESCRIPTION
**Pull request by ldjebran**
cover https://bugzilla.redhat.com/show_bug.cgi?id=1361309
```console
(sat-6.3) dl@DL:~/projects/redhat/robottelo$ pytest -v tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_register_twice_with_uppercase_chars_in_hostname
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.1.3, py-1.4.34, pluggy-0.4.0 -- /home/dl/.pyenv/versions/sat-6.3/bin/python
cachedir: .cache
rootdir: /home/dl/projects/redhat/robottelo, inifile:
plugins: xdist-1.20.0, services-1.2.1, mock-1.6.2, forked-0.2, cov-2.5.1
collected 1 item 
2017-09-20 16:24:44 - conftest - DEBUG - Found WONTFIX in decorated tests ['1156555', '1269196', '1378009', '1245334', '1217635', '1226425', '1147100', '1311113', '1199150', '1204686', '1221971', '1103157', '1230902', '1230865', '1214312', '1079482']

2017-09-20 16:24:44 - conftest - DEBUG - Collected 1 test cases


tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_register_twice_with_uppercase_chars_in_hostname PASSED

================================================== 0 tests deselected ==================================================
============================================== 1 passed in 304.08 seconds ==============================================
```

### CODE REVIEW

**Comment by oshtaier** [comment]
and it should not be `lower()` and then `upper()`, just normal name and then `upper()`

**Comment by ldjebran** [comment]
@oshtaier please point to the description that host name should be changed between registrations, 
add an other test for your case , but the bug it self does not say that
it say because  sat register hostname in lowercase it thinks that the host in an other host, raise error that that name is already taken

**Comment by ldjebran** [comment]
@oshtaier the bug description 
```
If a host has a hostname with any uppercase letters, re-registration will try to find it, but not be able to because its stored with lowercase letters.
```
is there some think I cannot understand here ??
bug steps:
```
Steps to reproduce:
1) hostname 'FooBar.example.com'
2) subscription-manager register
3) subscription-manager register --force
```
where changing hostname

**Comment by ldjebran** [comment]
@oshtaier I do not agree with you.
 if you want an other test that cover your scenario , no problem.

**Comment by oshtaier** [comment]
Okay, if you want to do in that way then why do you do `target_image = gen_string('alpha').upper()`?, use similar name to one in example: `'MyTest.test.com'`?

**Comment by ldjebran** [comment]
Ok no problem with the name, I will change that to some thing like AbCdEf

**Comment by oshtaier** [comment]
Thanks!

**Comment by ldjebran** [comment]
@oshtaier  updated the name generated for test was "KfMtWeXaHa"
```
(sat-6.3) dl@DL:~/projects/redhat/robottelo$ pytest -v tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_register_twice_with_uppercase_chars_in_hostname
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.1.3, py-1.4.34, pluggy-0.4.0 -- /home/dl/.pyenv/versions/sat-6.3/bin/python
cachedir: .cache
rootdir: /home/dl/projects/redhat/robottelo, inifile:
plugins: xdist-1.20.0, services-1.2.1, mock-1.6.2, forked-0.2, cov-2.5.1
collected 1 item 
2017-09-20 19:15:21 - conftest - DEBUG - Found WONTFIX in decorated tests ['1156555', '1269196', '1378009', '1245334', '1217635', '1226425', '1147100', '1311113', '1199150', '1204686', '1221971', '1103157', '1230902', '1230865', '1214312', '1079482']

2017-09-20 19:15:21 - conftest - DEBUG - Collected 1 test cases


tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_register_twice_with_uppercase_chars_in_hostname PASSED

================================================== 0 tests deselected ==================================================
============================================== 1 passed in 285.09 seconds ==============================================
```

File: tests/foreman/cli/test_host.py [hunk 1]
```diff
@@ -660,6 +660,62 @@ def test_negative_register_twice(self):
...
+        name = gen_string('alpha')
+        # make all the odd chars as upper case and the even one lower case
+        # to have a name like OpQrSTuVw
+        name_chars = list(name)
+        for i in range(len(name_chars)):
+            if i % 2 == 0:
+                name_chars[i] = name_chars[i].upper()
+            else:
+                name_chars[i] = name_chars[i].lower()
+        target_image = ''.join(name_chars)
```
Comment (abalakh) [inline tests/foreman/cli/test_host.py]: One-linear suggestion instead all these lines with conditions and double conversion:
```python
''.join((str.upper,str.lower)[i%2](ch) for i,ch in enumerate(name))
```
But maybe it's a bit too tricky, i'm not sure. Anyhow, not a blocker 🙂

Comment (abalakh) [reply tests/foreman/cli/test_host.py]: Or at least you could drop `else:`:
```python
name_chars = list(name.lower())
for i in range(len(name_chars)):
    if i % 2 == 0:
        name_chars[i] = name_chars[i].upper()
target_image = ''.join(name_chars)
```

**Comment by ldjebran** [comment]
@abalakh @oshtaier all comments addressed

generated host name was  : YvWcSbBrWk
```
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ pytest -v tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_register_twice_with_uppercase_chars_in_hostname
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.1.3, py-1.4.34, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.3.0/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.20.0, services-1.2.1, mock-1.6.2, forked-0.2, cov-2.5.1
collected 1 item 
2017-09-25 11:33:16 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/cli/test_host.py::HostCreateTestCase::test_positive_register_twice_with_uppercase_chars_in_hostname PASSED

============================================== 1 passed in 303.77 seconds ==============================================
```

**Pull request merged**